### PR TITLE
Feature/frontend

### PR DESCRIPTION
**Pull request by kierana**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+google-api-core==1.20.0
+google-api-python-client==1.9.3
+google-auth==1.17.2
+google-auth-httplib2==0.0.3
+google-cloud-core==1.3.0
+google-cloud-firestore==1.7.0
+google-cloud-pubsub==1.6.0
+googleapis-common-protos==1.52.0
+grpc-google-iam-v1==0.12.3
+grpcio==1.29.0
```
Comment (jcox-dev) [inline requirements.txt:17]: `grpcio` needs to come top of the list due to a versioning issue with Google API.

I'll tidy this file up as there's lots in here that might be sub and/or dev dependencies.

Comment (jcox-dev) [reply requirements.txt]: @kierana I've created a separate `dev-requirements.txt` to make it easier for running things locally while keeping the production environment / container clean.

File: app/blueprints/api.py [hunk 2]
```diff
@@ -28,8 +29,12 @@ def entry_list():
 
         if request.args.get('first_letter'):
             entries = entries.where('name_first_letter', '==', request.args.get('first_letter').lower())
-
+    
         results = [entry.to_dict() for entry in entries.get()]
+
+        if request.args.get('random'):
+            results = random.sample(results, int(request.args.get('random')))
```
Comment (jcox-dev) [inline app/blueprints/api.py:36]: @kierana as discussed random will need cache-busting (based upon time?)

File: requirements.txt (hunk 1 above)
Comment (kierana) [reply requirements.txt]: wicked, thanks

File: app/blueprints/api.py (hunk 2 above)
Comment (kierana) [reply app/blueprints/api.py]: I do this frontend - As the cache key is a URL, adding a timestamp to the URL breaks out the cache

**Pull request merged**